Sources/SwiftParserDiagnostics/SyntaxExtensions.swift

```diff
@@ -40,12 +40,13 @@ extension UnexpectedNodesSyntax {
     }
   }
 
-  /// If this only contains two tokens, one satisfying `firstCondition`, and one satisfying `secondCondition`,
-  /// return these tokens as a tuple, otherwise return `nil`. The function is not order sensitive
-  func twoTekens(firstSatisfying firstCondition: (TokenSyntax) -> Bool, secondSatisfying secondCondition: (TokenSyntax) -> Bool) -> (first: TokenSyntax, second: TokenSyntax)? {
+  /// If this only contains two tokens, the first satisfying `firstCondition`, and the second satisfying `secondCondition`,
+  /// return these tokens as a tuple, otherwise return `nil`.
+  func twoTokens(firstSatisfying firstCondition: (TokenSyntax) -> Bool,
+                 secondSatisfying secondCondition: (TokenSyntax) -> Bool) -> (first: TokenSyntax, second: TokenSyntax)? {
     let firstTokens = tokens(satisfying: firstCondition)
     let secondTokens = tokens(satisfying: secondCondition)
-    if firstTokens.count == 1, secondTokens.count == 1, firstTokens.first != secondTokens.first {
+    if firstTokens.count == 1, secondTokens.count == 1, self.first?.as(TokenSyntax.self) == firstTokens.first, firstTokens.first != secondTokens.first {
       return (firstTokens[0], secondTokens[0])
     } else {
       return nil
```

Comment:
All of this seem fairly roundabout. Wouldn’t it be simpler to do something like the following?

```swift
let sourceAccurateChildren = self.children(viewMode: .sourceAccurate)
guard sourceAccurateChildren.count == 2 else {
  return nil
}
guard firstCondition(sourceAccurateChildren[0]) && secondCondition(sourceAccurateChildren[1]) else {
  return nil
}
return (sourceAccurateChildren[0], sourceAccurateChildren[1])
```